apps/redis.go

```diff
@@ -17,8 +17,49 @@ package apps
 import (
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
 )
 
+type MetricsReceiverRedis struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,url"`
```

Comment:
This is being validated as a URL but the default is `localhost:6379`. Which is it, a URL or a host:port?

Reply:
That makes sense. I don't think it makes sense to document a default that doesn't have a scheme if we're intending this to be interpreted as a URL. I'd suggest that we update the documented default to "redis://localhost:6379" and documenting it as "Redis [URL](https://www.iana.org/assignments/uri-schemes/prov/redis). If the scheme is unspecified, `redis` is assumed." WDYT?

I just took a quick survey of some of the clients (geez, Redis has lot!?) and they seem super inconsistent about how they handle this, BTW:

- Redigo has an `address` [parameter](https://pkg.go.dev/github.com/gomodule/redigo/redis#Dial) that takes `host:port`, and an alternate API for parsing `redis://` URLs
- go-redis has an `Addr` [field](https://pkg.go.dev/github.com/go-redis/redis/v8#Options) that takes `host:port` with no option for URL that I can see
- aioredis has a `from_url` [method](https://aioredis.readthedocs.io/en/latest/api/high-level/#aioredis.client.Redis.from_url) that takes a URL, and no option for `host:port` that I see
- redis-py has `host` and `port` [parameters](https://redis-py.readthedocs.io/en/stable/#redis.Connection), and an alternate `from_url` method that takes a URL
- Walrus only documents `host` and `port` [parameters](https://walrus.readthedocs.io/en/latest/getting-started.html)
- redis-rs takes a `params` [argument](https://docs.rs/redis/0.21.3/redis/#connection-parameters) which can be cast from a URL string or be an instance of a struct with host and port

Reply:
Okay. Can we also name the field `address` to be more consistent with all the client libraries, since it doesn't accept a URL?

What about SSL and Unix sockets, since we don't have the scheme to set those kinds of connections?

Reply:
Looks like the requested changes have been applied. Thanks!